wallet/src/de/schildbach/wallet/ui/WalletFragment.kt

```diff
@@ -0,0 +1,342 @@
+            syncStatusTitle.setText(R.string.sync_status_sync_title)
+            syncStatusMessage.setText(R.string.sync_status_sync_completed)
+            updateSyncPaneVisibility(R.id.sync_status_pane, false)
+        } else {
+            syncPercentageView.setTextColor(resources.getColor(R.color.dash_gray))
+            updateSyncPaneVisibility(R.id.sync_status_pane, true)
+            syncStatusTitle.setText(R.string.sync_status_syncing_title)
+            syncStatusMessage.setText(R.string.sync_status_syncing_sub_title)
+        }
+    }
+
+    private fun initView() {
+        initQuickActions()
+        if (requireActivity() is OnSelectPaymentTabListener) {
+            pay_btn.setOnClickListener(View.OnClickListener {
+                (requireActivity() as OnSelectPaymentTabListener).onSelectPaymentTab(ACTIVE_TAB_PAY)
+            })
+            receive_btn.setOnClickListener {
+                (requireActivity() as OnSelectPaymentTabListener).onSelectPaymentTab(ACTIVE_TAB_RECEIVE)
+            }
+        }
+
+    }
+
+    private fun initViewModel() {
+        //
+        // Currently this is only used to check the status of Platform before showing
+        // the Join DashPay (evolution) button on the shortcuts bar.
+        // If that is the only function that the platform required for, then we can
+        // conditionally execute this code when a username hasn't been registered.
+        dashPayViewModel = ViewModelProvider(this)[DashPayViewModel::class.java]
+        dashPayViewModel!!.isPlatformAvailableLiveData.observe(viewLifecycleOwner, object : Observer<Resource<Boolean?>?> {
+            override fun onChanged(status: Resource<Boolean?>?) {
+                if (status?.status === Status.SUCCESS) {
+                    if (status.data != null) {
+                        isPlatformAvailable = status.data
+                    }
+                } else {
+                    isPlatformAvailable = false
+                }
+                showHideJoinDashPayAction()
+            }
+        })
+        AppDatabase.getAppDatabase().blockchainIdentityDataDao().loadBase().observe(viewLifecycleOwner,
+                Observer<BlockchainIdentityBaseData?> {
+                    fun onChanged(blockchainIdentityData: BlockchainIdentityBaseData) {
+                        if (blockchainIdentityData != null) {
+                            noIdentityCreatedOrInProgress = blockchainIdentityData.creationState == BlockchainIdentityData.CreationState.NONE
+                            showHideJoinDashPayAction()
+                            if (retryCreationIfInProgress && blockchainIdentityData.creationInProgress) {
+                                retryCreationIfInProgress = false
+                                activity?.startService(createIntentForRetry(requireActivity(), false))
+                            }
+                        }
+                    }
+                })
```

Comment:
Android Studio warns that onChange is never called and this observer never triggered the "Join DashPay" button to disappear since an identity was never detected.  When I removed `onChange` as follows, then the button was not visible when in identity/username existed.

```
        AppDatabase.getAppDatabase().blockchainIdentityDataDao().loadBase().observe(viewLifecycleOwner,
                Observer {
                        if (it != null) {
                            noIdentityCreatedOrInProgress = it.creationState == BlockchainIdentityData.CreationState.NONE
                            showHideJoinDashPayAction()
                            if (retryCreationIfInProgress && it.creationInProgress) {
                                retryCreationIfInProgress = false
                                activity?.startService(createIntentForRetry(requireActivity(), false))
                            }
                        }

                }
        )
```

Reply:
Fixed on https://github.com/dashevo/dash-wallet/pull/462/commits/eb3075fd9f97b15835c06a4358ddcb26df6c102d